extending/python_requires.rst

```diff
@@ -1,160 +1,245 @@
+      so they do not conflict with each other, but it is the responsibility of the user to keep consistency.
+    - They are not overridden from downstream consumers. Again, as they are private, they are not affected by other packages,
+      even consumers
+
+
+Reuse python sources
+--------------------
+
+In the example proposed we are using two functions through the ``base``
+variable: ``base.get_conanfile()`` and ``base.get_version()``. The first one is defined
+directly in the *conanfile.py* file, but the second one is in a different source file that
+we have exported together with the ``pyreq/version@user/channel`` recipe using the
+``exports`` attribute.
+
+This works without any Conan magic, it is just plain Python and you can even return a
+class from a function and inherit from it. That's just what we are proposing in this
+example: all the business logic in contained in the *Python Requires* so every recipe
+will reuse it automatically. The consumer only needs to define the ``name`` and ``version``:
 
 .. code-block:: python
 
     from conans import ConanFile, python_requires
-    
-    base = python_requires("MyBuild/0.1@user/channel")
 
-    class PkgTest(ConanFile):
-        ...
-        def build(self):
-            base.my_build(self.settings)
 
+    base = python_requires("pyreq/version@user/channel")
+
+    class ConsumerConan(base.get_conanfile()):
+        name = "consumer"
+        version = "version"
+
+        # Everything else is inherited
 
-Version ranges are possible with the version ranges notation ``[]``, similar to regular requirements.
-Multiple ``python_requires()`` are also possible
+while all the functional code is defined in the *python requires* recipe file:
 
 .. code-block:: python
-    :caption: **conanfile.py**
 
-    from conans import python_requires
-    
-    base = python_requires("MyBase/[~0.1]@user/channel")
-    other = python_requires("Other/1.2@user/channel")
+    from conans import ConanFile, python_requires
 
-    class Pkg(base.MyBase):
-        def source(self):
-            other.some_function()
+    [...]
 
-It is possible to structure the code in different files too:
+    def get_conanfile():
 
-.. code-block:: python
-    :caption: **conanfile.py**
+        class BaseConanFile(ConanFile):
+            def source(self):
+                [...]
 
-    from conans import ConanFile
-    import mydata # reuse the strings from here
-    class MyConanfileBase(ConanFile):
-        exports = "*.py"
-        def source(self):
-            self.output.info(mydata.src)
+            def build(self):
+                [...]
 
-.. code-block:: python
-    :caption: **mydata.py**
 
-    src = "My cool source!"
-    build = "My cool build!"
-    pkg = "My cool package!"
-    info = "My cool package_info!"
+Reuse source files
+------------------
+
+Up to now, we have been reusing python code, but we can also package files within the
```

Comment:
Really? You are kind of reusing the *scm_utils.py* in the previous example...

Reply:
:unamused: ...for a C++ developer, python code are not **sources** 🤣 

I understand your concern, maybe it is the difference between `exports` and `exports_sources`, up to here I've been reusing code from the `exports` family, now I'm targeting the `exports_sources` files. If you can suggest names for those sections...

Reply:
ok, np